### ISSUE TITLE
Import from tib

### ISSUE THREAD

**Issue by galvez** [pimlie/tib#1]
<img width="1160" alt="Screen Shot 2019-03-20 at 3 45 08 AM" src="https://user-images.githubusercontent.com/12291/54664660-a6430900-4ac3-11e9-8d87-b1c1d4686433.png">

Having a hard time getting this to work. Using same babel settings as the src.

**Comment by pimlie** [comment]
Hmm yes, I have been testing the module by using `yarn link` and that works with the posted babel config. But indeed when you install the module from npm the babel config doesnt work, not sure why there is a difference. Will have a look at it, a cjs build is also on my todo list btw.

**Comment by pimlie** [comment]
Ah wait, jest also has a [`transformIgnorePatterns`](https://jestjs.io/docs/en/configuration#transformignorepatterns-array-string) config option we need to set:
```js
  // jest.config.js
  transformIgnorePatterns: [
    '/node_modules/(?!(tib))/'
  ],
```

**Comment by galvez** [comment]
@pimlie nice, that worked. now I get:

```
  ● puppeteer › load page

    Error occured while loading 'puppeteer' browser: Error: Not supported
```

**Comment by galvez** [comment]
@pimlie I'm on Mac OS 10.12.4

**Comment by pimlie** [comment]
@galvez That probably means that Jest/Babel doesnt know what to do with the dynamic imports. Do you have the dynamic import plugin loaded? Also sometimes it helps to have a babel.config.js instead of babelrc due to some internal Babel behaviour differences.

**Comment by galvez** [comment]
@pimlie indeed, that was missing. thanks! now I'm getting:

```
    BrowserError: Error: PuppeteerBrowser: Could not find a chrome executable
```

I installed `puppeteer-core` and `chromedriver` -- is there anything else needed?

**Comment by galvez** [comment]
I think this is probably failing for my Mac OS version:

```
    execSync(
      `${LSREGISTER} -dump` +
        " | grep -i '(google chrome\\( canary\\)\\?|chromium).app$'" +
        ' | awk \'{$1=""; print $0}\''
    )
```

**Comment by galvez** [comment]
For reference, this is the result here:

```
	path:          /Applications/Google Chrome.app/Contents/Versions/61.0.3163.100/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Volumes/Macintosh HD/.Trashes/501/Applications/Google Chrome Canary.app
	name:          Chrome
	displayName:   Google Chrome
	identifier:    com.google.Chrome.canary (0x80007b78)
	canonical id:  com.google.chrome.canary (0x80007b77)
	executable:    Contents/MacOS/Google Chrome Canary
    CFBundleDisplayName = "Google Chrome";
    CFBundleExecutable = "Google Chrome Canary";
    CFBundleIdentifier = "com.google.Chrome.canary";
    CFBundleName = Chrome;
	activityTypes: ua-9c0e37c2deb108, pv-f3f9067b08e0ce, NSUserActivityTypeBrowsingWeb, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.canary
	path:          /Volumes/Macintosh HD/.Trashes/501/Applications/Google Chrome.app
	name:          Chrome
	displayName:   Google Chrome
	identifier:    com.google.Chrome (0x800026b3)
	canonical id:  com.google.chrome (0x800026b2)
	executable:    Contents/MacOS/Google Chrome
    CFBundleDisplayName = "Google Chrome";
    CFBundleExecutable = "Google Chrome";
    CFBundleIdentifier = "com.google.Chrome";
    CFBundleName = Chrome;
	activityTypes: pv-cb76a9f0433f1b, ua-9c0e37c2deb108, NSUserActivityTypeBrowsingWeb, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome
	path:          /Volumes/Macintosh HD/.Trashes/501/Applications/Google Chrome2.app
	name:          Chrome
	displayName:   Google Chrome
	identifier:    com.google.Chrome (0x800026b3)
	canonical id:  com.google.chrome (0x800026b2)
	executable:    Contents/MacOS/Google Chrome
    CFBundleDisplayName = "Google Chrome";
    CFBundleExecutable = "Google Chrome";
    CFBundleIdentifier = "com.google.Chrome";
    CFBundleName = Chrome;
	activityTypes: pv-cb76a9f0433f1b, ua-9c0e37c2deb108, NSUserActivityTypeBrowsingWeb, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome
	path:          /Applications/Google Chrome.app/Contents/Versions/59.0.3071.115/Google Chrome Framework.framework/Versions/A/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/59.0.3071.115/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Volumes/Google Chrome/Google Chrome.app
	name:          Chrome
	displayName:   Google Chrome
	identifier:    com.google.Chrome (0x800026b3)
	canonical id:  com.google.chrome (0x800026b2)
	executable:    Contents/MacOS/Google Chrome
    CFBundleDisplayName = "Google Chrome";
    CFBundleExecutable = "Google Chrome";
    CFBundleIdentifier = "com.google.Chrome";
    CFBundleName = Chrome;
	activityTypes: pv-cb76a9f0433f1b, ua-9c0e37c2deb108, NSUserActivityTypeBrowsingWeb, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome
	path:          /Applications/Google Chrome.app/Contents/Versions/57.0.2987.133/Google Chrome Framework.framework/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/57.0.2987.133/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Volumes/Google Chrome/Google Chrome.app/Contents/Versions/57.0.2987.133/Google Chrome Framework.framework/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Volumes/Google Chrome/Google Chrome.app/Contents/Versions/57.0.2987.133/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Applications/Google Chrome.app/Contents/Versions/58.0.3029.81/Google Chrome Framework.framework/Versions/A/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/58.0.3029.81/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Applications/Google Chrome.app/Contents/Versions/60.0.3112.90/Google Chrome Framework.framework/Versions/A/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/60.0.3112.90/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Applications/Google Chrome.app/Contents/Versions/58.0.3029.96/Google Chrome Framework.framework/Versions/A/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/58.0.3029.96/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Applications/Google Chrome.app/Contents/Versions/60.0.3112.101/Google Chrome Framework.framework/Versions/A/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/60.0.3112.101/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Applications/Google Chrome.app
	name:          Chrome
	displayName:   Google Chrome
	identifier:    com.google.Chrome (0x800026b3)
	canonical id:  com.google.chrome (0x800026b2)
	executable:    Contents/MacOS/Google Chrome
    CFBundleDisplayName = "Google Chrome";
    CFBundleExecutable = "Google Chrome";
    CFBundleIdentifier = "com.google.Chrome";
    CFBundleName = Chrome;
	activityTypes: pv-cb76a9f0433f1b, ua-9c0e37c2deb108, NSUserActivityTypeBrowsingWeb, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome
    "com.apple.application-identifier" = "EQHXZ8M8AV.com.google.Chrome";
        "EQHXZ8M8AV.com.google.Chrome.webauthn"
	path:          /Users/jonas/Dropbox/Notebook/Work/checkout/node_modules/puppeteer/.local-chromium/mac-637110/chrome-mac/Chromium.app
	path:          /Users/jonas/slides/node_modules/puppeteer/.local-chromium/mac-637110/chrome-mac/Chromium.app
	path:          /Applications/Google Chrome.app/Contents/Versions/60.0.3112.113/Google Chrome Framework.framework/Versions/A/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/60.0.3112.113/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Applications/Google Chrome.app/Contents/Versions/58.0.3029.110/Google Chrome Framework.framework/Versions/A/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/58.0.3029.110/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Applications/Google Chrome.app/Contents/Versions/61.0.3163.79/Google Chrome Framework.framework/Versions/A/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/61.0.3163.79/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Applications/Google Chrome.app/Contents/Versions/61.0.3163.91/Google Chrome Framework.framework/Versions/A/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/61.0.3163.91/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Applications/Google Chrome.app/Contents/Versions/61.0.3163.100/Google Chrome Framework.framework/Versions/A/Resources/app_mode_loader.app
	identifier:    com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@ (0x800026cf)
	canonical id:  com.google.chrome.app.@app_mode_shortcut_id@ (0x800026ce)
    CFBundleIdentifier = "com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@";
	activityTypes: NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.app.@APP_MODE_SHORTCUT_ID@, pv-d1632667ddec4a
	path:          /Applications/Google Chrome.app/Contents/Versions/68.0.3440.106/Google Chrome Helper.app
	name:          Chrome Helper
	displayName:   Google Chrome Helper
	identifier:    com.google.Chrome.helper (0x800026d8)
	canonical id:  com.google.chrome.helper (0x800026d7)
	executable:    Contents/MacOS/Google Chrome Helper
    CFBundleDisplayName = "Google Chrome Helper";
    CFBundleExecutable = "Google Chrome Helper";
    CFBundleIdentifier = "com.google.Chrome.helper";
    CFBundleName = "Chrome Helper";
	activityTypes: pv-e52d6aad5e536f, NOTIFICATION#EQHXZ8M8AV:com.google.Chrome.helper
	path:          /Users/jonas/Projects/FLAGCX/flagcx-app/node_modules/puppeteer/.local-chromium/mac-536395/chrome-mac/Chromium.app
	path:          /Users/jonas/Projects/FLAGCX/flagcx-app/node_modules/puppeteer/.local-chromium/mac-536395/chrome-mac/Chromium.app/Contents/Versions/66.0.3347.0/Chromium Framework.framework/Versions/A/Resources/app_mode_loader.app
	path:          /Users/jonas/Projects/FLAGCX/flagcx-app/node_modules/puppeteer/.local-chromium/mac-536395/chrome-mac/Chromium.app/Contents/Versions/66.0.3347.0/Chromium Helper.app
	path:          /Volumes/Google Chrome
	disk image:    /Users/jonas/Downloads/googlechrome.dmg
	all roles:     com.google.chrome (0x800026b2)
	all roles:     com.google.chrome (0x800026b2)
	all roles:     com.google.chrome (0x800026b2)
```

**Comment by pimlie** [comment]
@galvez The chrome detector is not something I made myself but is copied from Nuxt and I think its orginally from Google self. So I am afraid I cant help you with that

You could just install puppeteer (not -core), that should already contain a Chrome executable if the detector cannot find it. But if you know the path yourself, you can also set `PUPPETEER_EXECUTABLE_PATH` in your env with the path to your Chrome executable as value.

Btw, I just released `v0.2` which contains a commonjs build so you shouldnt have to worry about Babel configs anymore
